python/ray/serve/controller.py

```diff
@@ -846,6 +846,8 @@ def deploy_serve_application(
         name: application name. If specified, application will be deployed
             without removing existing applications.
         route_prefix: route_prefix. Define the route path for the application.
+    Returns:
+        Returns None if no error is raised. Otherwise, returns error message.
     """
     try:
         from ray import serve
```

Comment:
why are these delayed imports here btw?

Reply:
We can probably move `call_app_builder_with_args_if_necessary` out of the task, but moving the others would cause the controller to depend on `api.py` (to access `serve.run()` and `serve.build()`) which seems circular.